### PR TITLE
Add NodeJS prompting sample

### PR DESCRIPTION
**Pull request by rokulka**
Add NodeJS prompting sample
Update Readme file

### CODE REVIEW

File: experimental/qnamaker-prompting/javascript_nodejs/dialogs/functionDialogBase.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    getPersistedState(dialogInstance) {
+        return dialogInstance.state[functionStateName];
+    }
+
+    async runStateMachine(dc) {
+       
+        var oldState = this.getPersistedState(dc.activeDialog);
+                
+        var input = dc.context.activity.text;
```
Comment (johnataylor) [inline experimental/qnamaker-prompting/javascript_nodejs/dialogs/functionDialogBase.js]: this line isn't necessary...

and just took a quick look at the C# and the same issue is there - can you remove it from both.

File: experimental/qnamaker-prompting/javascript_nodejs/bots/qnaBot.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { MessageFactory } = require('botbuilder');
+const { DialogBot } = require('./dialogBot');
+
+class QnABot extends DialogBot {
+    constructor(conversationState, userState, dialog, logger) {
```
[the remark this answers is not in the document]
Comment (rokulka) [reply experimental/qnamaker-prompting/javascript_nodejs/bots/qnaBot.js]: I'd borrowed this pattern from the Core-Bot sample. But yeah, it makes sense, I've udpated it.

File: experimental/qnamaker-prompting/javascript_nodejs/dialogs/functionDialogBase.js (hunk 1 above)
Comment (rokulka) [reply experimental/qnamaker-prompting/javascript_nodejs/dialogs/functionDialogBase.js]: Done.

File: experimental/qnamaker-prompting/javascript_nodejs/bots/dialogBot.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            await next();
+        });
+
+        this.onDialog(async (context, next) => {
+
+            // Save any state changes. The load happened during the execution of the Dialog.
+            await this.conversationState.saveChanges(context, false);
+            await this.userState.saveChanges(context, false);
+
+            next();
```
[the remark this answers is not in the document]
Comment (rokulka) [reply experimental/qnamaker-prompting/javascript_nodejs/bots/dialogBot.js]: Done.

File: experimental/qnamaker-prompting/javascript_nodejs/bots/dialogAndWelcomeBot.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { DialogBot } = require('./dialogBot');
+
+class DialogAndWelcomeBot extends DialogBot {
```
[the remark this answers is not in the document]
Comment (rokulka) [reply experimental/qnamaker-prompting/javascript_nodejs/bots/dialogAndWelcomeBot.js]: Done.

**Pull request merged**